src/state/slices/portfolioSlice/selectors.ts

```diff
@@ -986,23 +986,48 @@ export const selectAssetEquityItemsByFilter = createDeepEqualOutputSelector(
       }
     })
     const staking = stakingOpportunities.map(stakingOpportunity => {
-      // Because the underlying assets can have different precisions
-      // We need to convert it to the asset we are viewing to get correct amounts to sum together.
-      const underlyingAssetPrecision =
-        assets[stakingOpportunity.assetId]?.precision ?? asset?.precision
-      const cryptoAmountBaseUnit = convertPrecision({
-        value: bnOrZero(stakingOpportunity.cryptoAmountBaseUnit),
-        inputExponent: underlyingAssetPrecision,
-        outputExponent: asset?.precision ?? 0,
-      }).toString()
-      const amountCryptoPrecision = fromBaseUnit(
-        bnOrZero(cryptoAmountBaseUnit),
-        asset?.precision ?? 0,
-      )
+      const { amountCryptoPrecision, fiatAmount } = (() => {
+        const underlyingAssetIndex = stakingOpportunity.underlyingAssetIds.findIndex(
+          assetId => assetId === asset?.assetId,
+        )
+        const underlyingAssetPrecision =
+          assets[stakingOpportunity.assetId]?.precision ?? asset?.precision
+
+        const totalCryptoAmountPrecision = fromBaseUnit(
+          bnOrZero(stakingOpportunity.cryptoAmountBaseUnit),
+          underlyingAssetPrecision ?? 0,
+        )
+
+        const amountCryptoBaseUnit = bnOrZero(totalCryptoAmountPrecision)
```

Comment:
Can we avoid going from base unit and back to precision then back to base unit here?

Reply:
Didn't change anything but naming so its clearer because we need to multiply the precions by the underlying asset base unit to get the final amount based on the ratio

In [4bade8f](https://github.com/shapeshift/web/pull/7644/commits/4bade8f9fe1ac3742ff2474ea6faa38eed2ad30e)